x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/planner/QueryFolderOkTests.java

```diff
@@ -112,7 +112,8 @@ private static void addSpec(ArrayList<Object[]> arr, String name, String query,
     public void test() {
         // skip tests that do not make sense from case sensitivity point of view
         boolean isCaseSensitiveValidTest = name.toLowerCase(Locale.ROOT).endsWith("-casesensitive") && configuration.isCaseSensitive()
-            || name.toLowerCase(Locale.ROOT).endsWith("-caseinsensitive") && configuration.isCaseSensitive() == false;
+            || name.toLowerCase(Locale.ROOT).endsWith("-caseinsensitive") && configuration.isCaseSensitive() == false
+            || name.toLowerCase(Locale.ROOT).endsWith("sensitive") == false;
```

Comment:
maybe move `name.toLowerCase(Locale.ROOT).endsWith("sensitive") == false` to beginning, short-circuit the rest?